models/articles-model.js

```diff
@@ -1,8 +1,23 @@
 const db = require("../db/connection")
 
 exports.fetchArticles = () => {
-    return db.query(`SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, CAST(COUNT(comments.article_id) AS int) AS comment_count FROM articles LEFT JOIN comments ON articles.article_id=comments.article_id GROUP BY articles.article_id;`)
-    .then((articlesArray) => {
-        return articlesArray.rows
+    return db
+    .query(`SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, CAST(COUNT(comments.article_id) AS int) AS comment_count FROM articles LEFT JOIN comments ON articles.article_id=comments.article_id GROUP BY articles.article_id;`)
+    .then((result) => {
+        return result.rows
+    })
+}
+
+exports.fetchArticleById = (article_id) => {
+    return db
+    .query(`SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, articles.body FROM articles WHERE article_id = $1`, [article_id])
+    .then((result) => {
+        if(result.rows.length === 0) {
+            // The following is related to "handleCustomerErrors" in "error-handling-controller.js":
+            return Promise.reject({message: 'Article id not found.', status: 404})
```

Comment:
Awesome way of checking if an article is valid - sending back an object with a message and status is great as this can just be sent straight from your custom error handler without any extra logic 👍

Reply:
I shall give the credit to Rob for this ingenious piece code.